onadata/apps/viewer/tasks.py

```diff
@@ -349,3 +351,39 @@ def delete_export(export_id):
         export.delete()
         return True
     return False
+
+
+@task.periodic_task(
+    run_every=(crontab(hour='*/7')),
+    ignore_result=True
+)
+def check_pending_exports():
+    """
+    Exports that have not completed within a set time should be marked as
+    failed
+    """
+    h = settings.EXPORT_TASK_LIFESPAN
+    time_threshold = datetime.now() - timedelta(hours=h)
+    exports = Export.objects.filter(internal_status=Export.PENDING,
+                                    created_on__lt=time_threshold)
+    for export in exports:
+        export.internal_status = Export.FAILED
+        export.save()
+    return True
+
+
+@task.periodic_task(
+    run_every=(crontab(hour='*/7')),
+    ignore_result=True
+)
+def delete_old_failed_exports():
+    """
+    Delete old failed exports
+    """
+    h = settings.EXPORT_TASK_LIFESPAN
+    time_threshold = datetime.now() - timedelta(hours=h)
```

Comment:
You should probably use the time zone aware `django.utils.timezone.now()`.
The `h` could probably be more descriptive like `task_lifespan`.